packages/pulse-core/lib/utils.ts

```diff
@@ -43,7 +43,7 @@ export function normalizeDeps(deps: Array<State> | State) {
 
 export const copy = val => {
   // ignore if primitive type
-  if (typeof val !== 'object') return val;
+  if (typeof val !== 'object' || val instanceof Date) return val;
```

Comment:
This seems like a very specific fix — I wonder if there are any other data types that run into this? I'll merge this, but would love to look further into this for other potential issues

Reply:
@hadenpf I sent a PR that solves this, #156